common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -364,18 +364,18 @@ public void record(Object hint) {
         }
 
         private void record0(Object hint, int recordsToSkip) {
-            if (creationRecord != null) {
+            if (creationRecord != null && MAX_RECORDS > 0) {
```

Comment:
I would prefer to keep the change. Maybe we should add extra comment or find a better way to handle `MAX_RECORDS == 0`;